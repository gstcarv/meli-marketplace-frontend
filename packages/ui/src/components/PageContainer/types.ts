import { HTMLProps } from '../../types';

export type PageContainerProps = HTMLProps<HTMLDivElement>;

export type PageContainerContentProps = HTMLProps<HTMLDivElement>;

export type PageContainerHeadingProps = HTMLProps<HTMLDivElement>;
