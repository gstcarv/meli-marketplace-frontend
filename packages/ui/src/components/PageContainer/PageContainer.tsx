import { PropsWithChildren } from 'react';

export const PageContainer = ({ children }: PropsWithChildren) => {
    return <>children</>;
};
