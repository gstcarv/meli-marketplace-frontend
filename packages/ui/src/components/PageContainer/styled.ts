import { breakpoints } from '@meli/design-tokens';
import React, { ComponentProps } from 'react';
import styled from 'styled-components';

export const Container = styled.div`
    margin: ${({ theme }) => theme.spacing.get('xl')} 10%;

    @media screen and (max-width: ${breakpoints.get('sm')}) {
        margin: ${({ theme }) => theme.spacing.get('xl')} 5%;
    }
` as React.FC<ComponentProps<'div'>>;

export const ContentContainer = styled.main`
    background-color: #ffffff;
    padding: ${({ theme }) => theme.spacing.get('md')};
    border-radius: ${({ theme }) => theme.radii.get('xs')};
` as React.FC<ComponentProps<'main'>>;

export const ContentHeader = styled.div`
    margin-bottom: ${({ theme }) => theme.spacing.get('md')};
` as React.FC<ComponentProps<'div'>>;
