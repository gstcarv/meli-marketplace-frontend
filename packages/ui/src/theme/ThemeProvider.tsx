import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
