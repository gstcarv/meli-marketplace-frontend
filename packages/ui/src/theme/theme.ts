import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
