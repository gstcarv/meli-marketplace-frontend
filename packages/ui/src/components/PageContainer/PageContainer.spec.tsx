import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { PageContainer } from './PageContainer';

describe('<PageContainer />', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <PageContainer>mock-page-content</PageContainer>
            </ThemeProvider>
        );

        expect(screen.getByText('mock-page-content')).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
