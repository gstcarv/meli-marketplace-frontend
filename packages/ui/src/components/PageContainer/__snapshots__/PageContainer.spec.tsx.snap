// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> should render correctly 1`] = `
<div>
  <div
    class="sc-blHHSb fbghbi"
  >
    <main
      class="sc-gtLWhw bXRjqK"
    >
      mock-page-content
    </main>
  </div>
</div>
`;

exports[`<PageContainer /> should render with heading 1`] = `
<div>
  <div
    class="sc-blHHSb fbghbi"
  >
    <div
      class="sc-egkSDF dekWDE"
    >
      mock-page-heading
    </div>
    <main
      class="sc-gtLWhw bXRjqK"
    >
      mock-page-content
    </main>
  </div>
</div>
`;
